massa-network/src/peer_info_database.rs

```diff
@@ -494,50 +460,35 @@ impl PeerInfoDatabase {
         if !ip.is_global() {
             return Err(NetworkError::InvalidIpError(*ip));
         }
-        let (available_bootstrap_conns, available_nonbootstrap_conns) =
-            self.get_available_out_connection_attempts();
+        let mut update_happened = false;
+        let peer_type = if let Some(p) = self.get_peer_type(ip) {
+            p
+        } else {
+            let mut p = PeerInfo::new(*ip, false);
+            p.active_out_connection_attempts += 1;
+            self.peers.insert(*ip, p);
+            update_happened = true;
+            self.get_peer_type(ip).unwrap()
+        };
+
+        self.update_global_active_out_connection_attempt_count(
+            peer_type,
+            true,
+            NetworkConnectionErrorType::TooManyConnectionAttempts(*ip),
+        )?;
+
         let peer = self.peers.get_mut(ip).ok_or({
             NetworkError::PeerConnectionError(NetworkConnectionErrorType::PeerInfoNotFoundError(
                 *ip,
             ))
         })?;
-        if peer.bootstrap {
-            if available_bootstrap_conns == 0 {
-                return Err(NetworkError::PeerConnectionError(
-                    NetworkConnectionErrorType::ToManyConnectionAttempt(*ip),
-                ));
-            }
-        } else if available_nonbootstrap_conns == 0 {
-            return Err(NetworkError::PeerConnectionError(
-                NetworkConnectionErrorType::ToManyConnectionAttempt(*ip),
-            ));
-        }
         peer.active_out_connection_attempts += 1;
```

Comment:
thisis also an update, and should set "update_happened" to true

Reply:
so if we reached that point there is always an update so no need for update_happened. I was thinking that we just call self.update() when an ip was added, removed or ip type update, not every time we update a counter ?

Reply:
if that's the case, yes